docs/sections/versions/baker-3-release-notes.md

```diff
@@ -0,0 +1,215 @@
+
+
+#### Decide on what to be notified on when firing the event
+Like in the old versions you can ask Baker to notify on specific moments.
+This can be when the event is received or when the processing is completed.
+New in this release is that you can now specify this when firing the event itself.
+The advantage is that Baker can optimize the resources it uses in those cases.
+This is done using the fireEventAndResolveWhenReceived and fireEventAndResolveWhenCompleted methods.
+
+```scala tab="Scala"
+val result: Future[SensoryEventStatus] = baker.fireEventAndResolveWhenReceived(recipeInstanceId, orderPlaced)
+val result: Future[SensoryEventResult] = baker.fireEventAndResolveWhenCompleted(recipeInstanceId, paymentMade)
+```
+
+```java tab="Java"
+CompletableFuture<SensoryEventStatus> result = baker.fireEventAndResolveWhenReceived(recipeInstanceId, paymentMade);
+CompletableFuture<SensoryEventResult> result = baker.fireEventAndResolveWhenCompleted(recipeInstanceId, firstOrderPlaced);
+```
+
+#### Wait for a specific Event
+A new moment you can ask Baker to notify you on is on a specific Event.
+You can use the fireEventAndResolveOnEvent method for this.
+
+```scala tab="Scala"
+val result: Future[SensoryEventResult] = baker.fireEventAndResolveOnEvent(recipeInstanceId, orderPlaced, "eventName")
+```
+
+```java tab="Java"
+CompletableFuture<SensoryEventResult> result =
+                baker.fireEventAndResolveOnEvent(recipeInstanceId, firstOrderPlaced, "eventName");
+```
+
+#### EventResult
+When firing a event and waiting for completion stage or waiting for a specific event a EventResult object will be returned.
+This EventResult object contains the SensoryEventStatus, event names and created ingredients.
+This can be useful if you need to make a decision depending on the state.
+In these cases there is no need to inquire on Baker itself anymore.
+
+```scala tab="Scala"
+case class SensoryEventResult(
+                               sensoryEventStatus: SensoryEventStatus,
+                               eventNames: Seq[String],
+                               ingredients: Map[String, Value]
+) extends common.SensoryEventResult with ScalaApi
+```
+
+```java tab="Java"
+case class SensoryEventResult(
+                               sensoryEventStatus: SensoryEventStatus,
+                               eventNames: java.util.List[String],
+                               ingredients: java.util.Map[String, Value]
+) extends common.SensoryEventResult with JavaApi {
+
+  def getSensoryEventStatus: SensoryEventStatus = sensoryEventStatus
+
+  def getEventName: java.util.List[String] = eventNames
+
+  def getIngredients: java.util.Map[String, Value] = ingredients
+}
+```
+
+### New EventListener functions
+For the EventListener we have moved away from the annotation based approach.
+In this version you can add a EventListener function instead.
+See the [event listener section](http://localhost:8000/sections/reference/event-listener/) for more details.
```

Comment:
"http://localhost:8000/sections/reference/event-listener/" will probably not work when deployed xD, needs a relative path